src/Compilers/Core/Portable/CommandLine/EditorConfig.SectionNameMatching.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal sealed partial class EditorConfig
+    {
+        /// <summary>
+        /// Takes a <see cref="Section.Name"/> and compiles the file glob
+        /// inside to a regex which recognizes the given language.
+        /// </summary>
+        public static string CompileSectionNameToRegEx(string sectionName)
+        {
+            // An editorconfig section name is a language for recognizing file paths
+            // defined by the following grammar:
+            //
+            // <path> ::= <path-list>
+            // <path-list> ::= <path-item> | <path-item> <path-list>
+            // <path-item> ::= "*"  | "**" | "?" | <char> | <choice> | <range>
+            // <char> ::= any unicode character
+            //
+            // PROTOTYPE(editorconfig): Below remains unimplemented
+            //
+            // <choice> ::= "{" <choice-list> "}" 
+            // <choice-list> ::= <path-list> | <path-list> "," <choice-list>
+            // <range> ::= "{" <integer> ".." <integer> "}"
+            // <integer> ::= "-" <digit-list> | <digit-list>
+            // <digit-list> ::= <digit> | <digit> <digit-list>
+            // <digit> ::= 0-9
+
+            var sb = new StringBuilder();
```

Comment:
>StringBuilder [](start = 25, length = 13)

`PooledStringBuilder.GetInstance()`?

Reply:
Same deal as the other PR -- this code may be shared. I'm staying away from Roslyn-specific stuff for now.

Reply:
Should we file a tracking bug or make a prototype comment to follow up if we determine that this won't be shared?

---
In reply to: [190944186](https://github.com/dotnet/roslyn/pull/27119#discussion_r190944186) [](ancestors = 190944186)

Reply:
Eh, since it's just perf I think we can address it as part of a larger perf investigation if it turns out to be important.

Reply:
i think we need at least a prototype comment to track this.

What is your non-roslyn envisioned usage? 

---
In reply to: [191026972](https://github.com/dotnet/roslyn/pull/27119#discussion_r191026972) [](ancestors = 191026972)